docs/configuration.md

```diff
@@ -542,17 +542,28 @@ Let's take the example of our matrix connector. Inside the module we set the con
 
 ```python
 from voluptuous import Required
-
-CONFIG_SCHEMA = {
-    Required("credentials"): dict,
-    Required("rooms"): dict,
-    "homeserver": str,
-    "nick": str,
-    "room_specific_nicks": bool,
-}
-```
-
-As you can see 'credentials` and `rooms` are required fields for this connector and we expect them to be dictionaries.
+CONFIG_SCHEMA = Any(
+    {
+        Required("mxid"): str,
+        Required("password"): str,
+        Required("rooms"): dict,
+        "homeserver": str,
+        "nick": str,
+        "room_specific_nicks": bool,
+    },
+    {
+        Required("token"): str,
+        Required("rooms"): dict,
+        "homeserver": str,
```

Comment:
If we are using token rather than mxid, we are actually going to have to require homeserver as we can't do a well-known lookup for it.